### ISSUE TITLE
[Feature] Implementing GH actions (GHA) as method to run K3ai from within the repo

### ISSUE THREAD

**Issue by alefesta** [k3ai/k3ai#19]
As an extension of #18, we should aim to develop a GHA that allows a user to operate directly from within its own repo.

**Comment by burntcarrot** [comment]
I'm working on it, I found some flaws:
- The PAT is **always read through user input**; there is no option of passing PAT through a flag.
- I tried piping the PAT to `k3ai up`, but it didn't work.

Hence, I'm working on writing a fix, which adds a `-p` flag for the PAT token. It works locally for me, I'll open up a PR on your approval.

This would potentially work with GitHub Actions, with the usage of `yes` to bypass the prompt wait. Here's how it works locally

```
❯ yes | go run main.go up -p "<GITHUB_TOKEN>"
🎉🎉🎉 Welcome to K3ai 🎉🎉🎉
📢      Give us a second and we will start the process...
 ❌     Missing GitHub Authentication Token, please paste it here:
 ✔️     GitHub Authentication Token: OK
 ❔     To avoid asking for the token everytime, are you ok if we save it as .env variable?
✔ Yes
 ✔️     Proceeding with configuration,please wait...

 🚀     Loading K3ai configuration...  100% |████████████████████████████████████████|    s:0s]
 🚀     Creating database...           100% |████████████████████████████████████████|    s:0s]
 🚀     Retrieving plugin list...        2% |                                        |  [0s:2s]
```

Feel free to respond to this after the weekend. My intention is not to annoy you. 😊

**Comment by alefesta** [comment]
So cool! Passing the PAT as a flag sound a good idea I was thinkin we may do a community session next week to talk about some of the ideas. And no worries I'm super happy you like to help on this project !!!

**Comment by burntcarrot** [comment]
If you want me to add this feature, could I open up a PR today?

**Comment by alefesta** [comment]
K3ai is for the people so yes anytime you like!!!

**Comment by burntcarrot** [comment]
**Update:**

I have set up a minimal example of setting up a `k3ai` cluster using Github Actions. You can see it action [here](https://github.com/burntcarrot/k3ai-actions-test/runs/4126233537?check_suite_focus=true).

*The current workflow doesn't use the binaries, as they don't have support for passing PAT tokens through the `-p` flag yet.*

**Comment by alefesta** [comment]
I like that my idea is o actually to read a confit in the repo to tell us what to install ( I.e. cluster X plus plug-in Y) so we may read the repo and excute the pipeline. Ideally grab the outcome and write it back in a folder in the repo.
So your GH is a great starting point !!! Amazing job

**Comment by burntcarrot** [comment]
Yes, I know about the "extracting the config" part. This was just a proof of concept; I was excited to try it on Github Actions! 😋

**Comment by alefesta** [comment]
And you did an amazing job... if we make this it will be awesome just think about be able to run a training job directly in GH actions !

**Comment by burntcarrot** [comment]
Keeping this open as we still have to deal with a complete end-to-end workflow. (Artifact collection support is missing)
